app/scripts/background.js

```diff
@@ -128,12 +148,17 @@ initialize().catch(log.error);
 /**
  * Initializes the MetaMask controller, and sets up all platform configuration.
  *
+ * @param {string} remotePort - remote application port connecting to extension.
  * @returns {Promise} Setup complete.
  */
-async function initialize() {
-  const initState = await loadStateFromPersistence();
-  const initLangCode = await getFirstPreferredLangCode();
-  await setupController(initState, initLangCode);
+async function initialize(remotePort) {
+  if (!initialState) {
+    initialState = await loadStateFromPersistence();
+  }
+  if (!initialLangCode) {
+    initialLangCode = await getFirstPreferredLangCode();
```

Comment:
breakout `loadStateFromPersistence` and `getFirstPreferredLangCode` into a function and use it above
inside of `if (process.env.ENABLE_MV3) {...}`

Reply:
I did some improvements here , plz check.